br/pkg/lightning/common/util.go

```diff
@@ -488,3 +492,119 @@ func GetAutoRandomColumn(tblInfo *model.TableInfo) *model.ColumnInfo {
 	}
 	return nil
 }
+
+// GetDropIndexInfos returns the index infos that need to be dropped and the remain indexes.
+func GetDropIndexInfos(
+	tblInfo *model.TableInfo,
+) (remainIndexes []*model.IndexInfo, dropIndexes []*model.IndexInfo) {
+	cols := tblInfo.Columns
+loop:
+	for _, idxInfo := range tblInfo.Indices {
+		if idxInfo.State != model.StatePublic {
```

Comment:
If any index is not public, this statement should be bu run. Otherwise, it may encounter the data-index inconsistent problem.

Reply:
Maybe we need add a precheck item, but not in this pr